target/linux/ath79/dts/ar9132_buffalo_wzr-hp-g300nh.dts

```diff
@@ -0,0 +1,259 @@
+// SPDX-License-Identifier: GPL-2.0-or-later OR MIT
+/dts-v1/;
+
+#include <dt-bindings/gpio/gpio.h>
+#include <dt-bindings/input/input.h>
+
+#include "ar9132.dtsi"
+
+/ {
+	compatible = "buffalo,wzr-hp-g300nh", "qca,ar9132";
+	model = "Buffalo WZR-HP-G300NH";
+
+	aliases {
+		led-boot = &led_diag;
+		led-failsafe = &led_security;
+		led-upgrade = &led_diag;
+	};
+
+	clock40mhz: ref {
+		compatible = "fixed-clock";
+		#clock-cells = <0>;
+		clock-frequency = <40000000>;
+	};
+
+	gpio2: gpio-extender {
+		compatible = "nxp,74hc153-gpio";
+		gpio-controller;
+		#gpio-cells = <2>;
+
+		// GPIOs used by this node
+		gpio-s0 = <&gpio 9 GPIO_ACTIVE_HIGH>;
+		gpio-s1 = <&gpio 11 GPIO_ACTIVE_HIGH>;
+		gpio-1y = <&gpio 12 GPIO_ACTIVE_HIGH>;
+		gpio-2y = <&gpio 14 GPIO_ACTIVE_HIGH>;
+	};
+
+	keys {
+		compatible = "gpio-keys-polled";
+		poll-interval = <20>;
+
+		aoss {
+			label = "aoss";
+			linux,code = <KEY_WPS_BUTTON>;
+			gpios = <&gpio2 0 GPIO_ACTIVE_LOW>;
+			debounce-interval = <60>;
+		};
+
+		reset {
+			label = "reset";
+			linux,code = <KEY_RESTART>;
+			gpios = <&gpio2 1 GPIO_ACTIVE_LOW>;
+			debounce-interval = <60>;
+		};
+/*
```

Comment:
There is the EV_SW property for switches:
https://github.com/openwrt/openwrt/blob/master/target/linux/ramips/dts/mt7621_buffalo_wsr-2533dhpl.dts#L88

Maybe try that in combination with the interrupt-driven driver. However, from what I remember EV_SW didn't really help much ...

Reply:
For some reason the fail safe is not being triggered now and I added all keys in the DTS. Probably there is something random going on but that EV_SW property might be a good idea to check if the issue remains.

That interrupt-driven driver just does not work. I suppose we would need to have these keys (GPIOs) to be connected to some interrupt controller but unfortunately they are not. It's just a GPIO-extender, whose input bank and pin is chosen with host GPIOs and it'll have to be polled.